### ISSUE TITLE
Need at least Pandoc 1.13 to convert from docx

### ISSUE THREAD

**Issue by bushchannelstepper** [ozmeka/ozmeka#40]
The production environment uses Centos which only has Pandoc 1.9 in repos.  To convert from docx we need at least 1.13

Pandoc's home page furnishes a 64 bit deb.  I tried installing this on dev with alien, but it lacks a dependency:

Error: Package: pandoc-1.13.2-2.x86_64 (/pandoc-1.13.2-2.x86_64)
           Requires: libgmp.so.10()(64bit)

We might need to build from source.

**Comment by DanielTosello** [comment]
I'll try to find a resolution for you in an external repo. Hopefully
compiling from source can be avoided... It does complicate the install a
bit.

**Comment by DanielTosello** [comment]
Scratch that it looks like it might be quite easy:

> yum install haskell-platform
> cabal update
> cabal install pandoc

Should do it apparently. Will test this locally in a few minutes.

**Comment by bushchannelstepper** [comment]
Thanks for the suggestions Daniel.  It looks like the version of Cabal on our Centos environment is too old to install the current version of Pandoc though.

**Comment by DanielTosello** [comment]
Yeah, i'm playing with it now to see if i can get around that. I'm using a similar env myself on my test box (Centos 6.5) so with any luck i will find you a solution that doesn't involve manual compilation.

**Comment by bushchannelstepper** [comment]
Thanks.  Paul's working on it here as well so don't sweat it.

**Comment by DanielTosello** [comment]
Get him to check https://halcyon.sh/ if he hasn't already ;) 

Confirmed Pandoc 1.13.2.1 installed on Centos 6.5 via Halcyon. Steps as follows: 

> - install Halcyon via script or github, e.g: 
> 
> eval "$( curl -sL https://github.com/mietek/halcyon/raw/master/setup.sh )" 
> - update Halcyon, ghc and cabal, set env requirements, etc.
> 
> /app/halcyon/halcyon install
> - install pandoc
> 
> /app/halcyon/halcyon install pandoc

Figured i might as well complete it since it was already underway. Install took about 30 mins for me...

**Comment by Paul-Nguyen** [comment]
The thing about halcyon is that it seems like overkill -- we just want to do the conversion and end up spinning up haskell base, cabal, sandbox, cabal update, pandoc deps, pandoc. I'd prefer rhel7/centos7 - at least we're up to a point where the system provided haskell and cabal are new enough to at least have ghc-deepseq > 1.3 provided, then it's just a matter of sandbox/cabal update/pandoc deps/pandoc.

**Comment by DanielTosello** [comment]
Yeah i get you. My priority was on ease for the deployer, but i guess ideally it'll be done via puppet or chef or similar... right?

**Comment by ptsefton** [comment]
We're putting this on hold. It doesn't really make sense to have a plugin
that is so hard to deploy - in the longer term will look at a simple plugin
that can talk to remote web services, then set up pandoc as one of those,
shared between repositories potentially.

For the Dharmae project will go with a short-term solution of using the API
to find and convert .doc files.

On Thu, May 14, 2015 at 9:12 AM, Daniel Tosello notifications@github.com
wrote:

> Yeah i get you. My priority was on ease for the deployer, but i guess
> ideally it'll be done via puppet or chef or similar... right?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ozmeka/ozmeka/issues/40#issuecomment-101845347.

## 

Peter Sefton +61410326955 pt@ptsefton.com http://ptsefton.com
Gmail, Twitter & Skype name: ptsefton
